### ISSUE TITLE
Best way to handle an IP change after `kexec`

### ISSUE THREAD

**Issue by arunoruto** [numtide/nixos-anywhere#415]
After the NixCon I was eager to try out nixos-anywhere and took an old mini PC so I can mess around a bit. It already had NixOS, but to make a real experiment, I installed Debian on it, so I can remotely distrohop.

After executing `nix run github:nix-community/nixos-anywhere -- --flake .#test root@a.b.c.d`, it was waiting for the host, but even after booting into nixos, it never reached it. A quick `ip addr` showed a different IP after the system was loaded, so of course it was unable to reach it again.

I could have maybe forced the same IP address somehow, but I ended up opening a new shell session and executing a new command `nix run github:nix-community/nixos-anywhere -- --flake .#kenpachi root@a.b.c.e -i /tmp/tmp.abcde12345/nixos-anywhere`.

`a.b.c.d` was the old IP, while `a.b.c.e` was the new one (can be found on your routers settings, the host name should be something like `nixos-installer`). I also had to provide the temporary SSH key for login `/tmp/tmp.abcde12345/nixos-anywhere`, which is displayed in the first lines during the execution of the script.

After fiddling around with this for an hour, it finally worked (I rarely use SSH keys, but it's a subject to change in the future!).
I am not sure if this has been mentioned somewhere in the documentation, so I wanted to ask if something like this should be also included? Maybe I am also mistaken and there is already a guide which slipped by me, while I was browsing the repo!

**Comment by josephfinlayson** [comment]
I also experienced this, and would like to know how to handle it!

**Comment by sedlund** [comment]
you can increase the dhcp lease time on your dhcp server or set a static lease with the mac address of the machine.  

or one could start with `--phases kexec` knowing that your network will change the ip on you.  then once you figure out your new ip, you can run the other phases with the new ip.

**Comment by Mic92** [comment]
For local machines where you can select boot mediums, I would recommend booting the NixOS images instead: https://github.com/nix-community/nixos-images?tab=readme-ov-file#iso-installer-images

It also has support for WiFi and avoids kexec, which may not work with hardware (i.e. some GPUs don't allow it).

**Comment by LarsOlt** [comment]
I fixed the issue by setting `networking.useDHCP = false;` in my nix configuration. After that I rebuilt my VPS with a fresh debian 12 install and ran nixos-anywhere again. Before I disabled DHCP my nixos got a different IP so I could no longer connect via ssh.

**Comment by n0rad** [comment]
I think the issue lives in the fact that the DHCP server is receiving after the kexec a different hostname, making it think it's better to give a new IP, instead of the previous one, even if the lease is still active and the mac address still the same.

I'm wondering if the hostname after kexec should be change to the same as before instead of being `nixos-installer`

**Comment by MaximilianCF** [comment]
Same thing with me. i'm getting nuts trying to set up NixOS in a Proxmox VM. I can reach the host using SSH, but after kexec runs the IP changes and so I'm not able to finish the process.
Even tweaking Proxmox to keep the IP static doesn't work at all!! DHCP is set to false in my configuration.nix
After rebooting on the VM, I go back to Debian which is the original OS I installed on the VM.

**Comment by sedlund** [comment]
> Do I need to replace the gateway and addresses with my own value? If so, how can I find those?

Yes, those are fake, and will be different.  GreenCloud has a couple different control panels, but the new one displays the required info in the networking tab of the host.  If not you'll have to boot the machine up or dhcp and record the info manually.  The ipv6 is only in the panel.

sorry for offtopic

**Comment by Wuppdich** [comment]
to add to the information here: i managed to get a working network on a Contabo VPS with this configuration: https://gist.github.com/Wuppdich/7df556df1966db076a3204ed3fe8c401
